pkg/medusa/refresh_secrets.go

```diff
@@ -52,7 +52,7 @@ func RefreshSecrets(dc *cassdcapi.CassandraDatacenter, ctx context.Context, clie
 		case recRes.IsRequeue():
 			requeues++
 			continue
-		case recRes.IsDone():
+		case !recRes.Completed():
```

Comment:
Changed this to keep the exact same behavior, but the overall function has issues: the block `if requeues > 0 { ... }` at line 58 is never reached (either with this version or the original ones), so retries are not behaving as expected.

Reply:
I think you're right, we should drop the continue statements for this to make sense.

Reply:
While there's a bug, it seems to be out of the scope of this PR.
Can we revert the change and file another issue for this?